test/integration-tests/IntegrationTests.Helpers/TestHelper.cs

```diff
@@ -82,9 +81,26 @@ public Container StartContainer(int traceAgentPort, int webPort)
 
         PowershellHelper.RunCommand($"docker exec {container.Name} curl -v {healthCheckEndpoint}", Output);
 
-        var (standardOutput, _) = PowershellHelper.RunCommand($"Write-Host (Test-NetConnection -ComputerName 'localhost' -Port {webPort}).TcpTestSucceeded", Output);
+        var healthChecksUrl = $"http://localhost:{webPort}/healthz";
+
+        // try healthz a few times
+        HttpStatusCode statusCode = default;
+        HttpClient client = new();
+
+        for (int retry = 0; retry < 5; retry++)
+        {
+            var response = await client.GetAsync(healthChecksUrl);
+            statusCode = response.StatusCode;
+
+            if (statusCode == HttpStatusCode.OK)
+            {
+                break;
+            }
+
+            Thread.Sleep(TimeSpan.FromSeconds(4));
```

Comment:
nit: since the code now already has an `await` use `Task.Delay` here instead of `Thread.Sleep`.